### PR TITLE
Refactor ruby, sass and testing for indidual files

### PR DESCRIPTION
**Pull request by joaojusto**
- Extracted ruby guides to its own file;
- Extracted sass guides to its own file;
- Extracted testing guides to its own file;
- Extracted templating guides to its own file;
- Moved ruby samples to ruby folder;
- Moved testing samples to testing folder;
- Move templating samples to templating folder;

### CODE REVIEW

**Comment by azevedo** [comment]
You've apparently moved a lot of files and changed others.
Could you please add to the description what you've done?

**Comment by joaojusto** [comment]
@azevedo-252 updated

File: style/README.md [hunk 1]
```diff
@@ -3,6 +3,11 @@ Style
 
 A guide for programming in style.
 
+* [JS](/js)
+* [Ruby](/ruby)
+* [SASS](/sass)
```
Comment (azevedo) [inline style/README.md]: Sass

Comment (azevedo) [inline style/README.md]: Actually it's not only here. Please change in the appropriate places

File: sass/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+SASS Style Guide
+=====
+### Formatting
+* Use the SASS syntax.
```
Comment (gabrielpoca) [inline sass/README.md]: Actually, I don't agree with this anymore. @azevedo-252 what about you?

Comment (azevedo) [inline sass/README.md]: That's a topic for a different PR, but in general, I'm more inclined for using the newer syntax: Scss

**Comment by joaojusto** [comment]
@azevedo-252 @gabrielpoca take a look now

File: style/README.md [hunk 3]
```diff
@@ -3,12 +3,13 @@ Style
 
 A guide for programming in style.
 
-Git
----
+In adition to this general guidelines below, we have the following more detailed, language/framework specific style guides:
```
Comment (azevedo) [inline style/README.md]: Review the english

File: style/README.md [hunk 4]
```diff
@@ -39,193 +40,3 @@ Naming
...
-* Avoid nesting within a media query.
-
-### Organization
-* Use Compass or Bourbon for a Sass library.
-* Use Normalize as a browser reset.
-* Use HTML structure for ordering of selectors. Don't just put styles at the
-  bottom of the Sass file.
-* Avoid having files longer than 100 lines.
-
-CoffeeScript
```
Comment (azevedo) [inline style/README.md:85]: You're just deleting and not replacing for anything else

File: style/README.md [hunk 5]
```diff
@@ -3,12 +3,13 @@ Style
...
-Git
----
+In adition to this general guidelines below, we have the following more detailed, language/framework specific style guides:
 
-* Avoid including files in source control that are specific to your
-  development machine or process.
-* Follow the [protocol](/protocol)
+* [JS](/js)
+* [Ruby](/ruby)
+* [Sass](/Sass)
```
Comment (azevedo) [inline style/README.md]: link doesn't exist

File: style/README.md [hunk 6]
```diff
@@ -39,193 +40,3 @@ Naming
...
-Email
------
-
-* Use one `ActionMailer` for the app. Name it `Mailer`.
-* Use the user's name in the `From` header and email in the `Reply-To` when
-  [delivering email on behalf of the app's users].
-
-[delivering email on behalf of the app's users]: http://robots.thoughtbot.com/post/3215611590/recipe-delivering-email-on-behalf-of-users
-
-Testing
```
Comment (azevedo) [inline style/README.md:186]: @zamith @naps62 This section is mostly ruby/rspec specific. Should we move this under ruby or something like that?

Comment (naps62) [inline style/README.md:186]: I think so

**Comment by azevedo** [comment]
lgtm :+1: 
@gabrielpoca @naps62 @zamith take a look as well

**Comment by gabrielpoca** [comment]
:+1: I don't believe we need the approval of everyone for this, he's just moving files around.

**Comment by naps62** [comment]
@joaojusto fix the menu on the main README, and merge both Ruby files

**Comment by naps62** [comment]
:ship:

File: sass/README.md (hunk 2 above)
Comment (zamith) [inline sass/README.md]: I much rather prefer the SASS syntax for almost all cases. I can think of maps, that don't look as good. But I think that's it.

In Rails projects, is very easy to have both, and use SCSS in a variables file, for instance, and SASS for the rest. I think that's what we did on easy.money, and I don't know of any reasons to complain about that approach.

Have you found any other use case in which SCSS might actually look nicer or be easier to read than SASS?

File: sass/README.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+* Use hyphens when naming mixins, extends, classes & variables: `span-columns` not `span_columns` or `spanColumns`.
+* Use space between property and value: `width: 20px` not `width:20px`.
+* Use a blank line above selector that has styles.
+* Prefer hex color codes `#000`.
+* Use `//` for comment blocks not `/* */`.
+* Use single or double quotes consistently. Preference is for double quotes, e.g., `content: ""`.
+* Use only lowercase, including colors.
```
Comment (zamith) [inline sass/README.md]: This has changed, with SUIT naming, correct?

File: sass/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+* Don't add a unit specification after `0` values, unless required by a mixin.
+
+### Order
+* Place scoped variables, @extends and @includes (excluding media query stuff) at the top of your declaration list, in that order.
+* Use alphabetical order for declarations.
```
Comment (zamith) [inline sass/README.md]: We should either enforce the use of this order on our projects, or change this to something else.

I find really "unnice" for each one to just go ahead and use the order of her/his preference. We should, as a team reach a consensus, and document it. 

We had this discussion before, and then the winner was this order, but then we see projects were this is not happening. We can have it again, if the arguments have changed, but going ahead and do what we feel like atm should be strongly discouraged, regardless of the style we, as a team, decide to go with.

File: sass/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+* Place media queries directly after the declaration list.
+* Place concatenated selectors second.
+* Place pseudo states and elements third.
+* Place nested selectors last.
+
+### Selectors
+* Don't use ID's for style.
+* Use meaningful names: `$visual-grid-color` not `$color` or `$vslgrd-clr`.
```
Comment (zamith) [inline sass/README.md]: @azevedo-252 Do you want to change this example to `$color-grid-visual`?

File: sass/README.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+* Don't nest more than 6 selectors deep.
+* Avoid using the HTML tag in the class name: `/*section*/.news` not `/*section*/.news-section`.
+* Avoid using HTML tags on classes with specific class names like `.featured-articles`.
+* Don't qualify selectors: `.widgets` not `div.widgets`.
+* Avoid nesting within a media query.
+
+### Organization
+* Use Compass or Bourbon for a Sass library.
+* Use Normalize as a browser reset.
+* Use HTML structure for ordering of selectors. Don't just put styles at the
```
Comment (zamith) [inline sass/README.md]: Do we want to add some CSS arch to this guides? I think we should reference one, even if we end up changing it in the future.

**Comment by zamith** [comment]
@groupbuddies/developers I commented on an outdated version of the pr... Had it open for a while. Damn AJAX! :p Please give a look at the comments, though.

**Comment by azevedo** [comment]
@zamith all those questions should be discussed in a totally different PR. @joaojusto merely moved things around

**Comment by zamith** [comment]
@azevedo-252 Not really another PR, since that would imply any code change, which I'm not proposing yet. We could comment files inline, but I guess this is as good of a way to do it as any.

If we arrive at any conclusion here, then we can create a new PR with those changes. What do you think?

**Comment by gabrielpoca** [comment]
@zamith please open an issue for this.

**Comment by azevedo** [comment]
I meant issue then

**Comment by zamith** [comment]
How can I open an issue to discuss specific lines?

**Comment by gabrielpoca** [comment]
You can reference lines from an issue.

**Pull request merged**